Building requires git repo

Users downloading the zip will get build failures since we directly reference git variables.

Here are some (might be others)
https://github.com/analogdevicesinc/iio-oscilloscope/blob/master/dialogs.c#L906
https://github.com/analogdevicesinc/iio-oscilloscope/blob/c37d0c2fa1e68dfdb301b4fef6cdc8974c91f102/dialogs.c#L946

The bug also exist if you use the Makefile or the CMakeLists.txt.

Building C object CMakeFiles/osc.dir/dialogs.c.o
/home/zynqdev/libiio-eclipse/osc-project/dialogs.c: In function ‘date_compare_against_build_date’:
/home/zynqdev/libiio-eclipse/osc-project/dialogs.c:906:21: error: too few arguments to function ‘atol’
  glong build_time = atol(GIT_COMMIT_TIMESTAMP);

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,12 @@ addons:
 matrix:
   fast_finish: true
   include:
+    - compile: gcc
+      os: linux
+      dist: trusty
+      env:
+        - LDIST=-trusty
+        - TEST_NON_GIT_REPO=1
     - compiler: gcc
       os: osx
       osx_image: xcode8.3

--- a/CI/travis/make
+++ b/CI/travis/make
@@ -8,6 +8,11 @@ if command -v brew ; then
 	export PATH="${QT_PATH}:$PATH"
 fi
 
+# Remove .git repo and see how it performs
+if [ "$TEST_NON_GIT_REPO" == "1" ] ; then
+	rm -rf .git
+fi
+
 mkdir -p build
 
 pushd build

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,7 @@ endif()
 
 # Get the GIT hash of the latest commit
 include(FindGit OPTIONAL)
-if (GIT_FOUND)
+if (GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
 	execute_process(
 		COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
 		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
@@ -58,9 +58,9 @@ if (GIT_FOUND)
 		)
 
 	set(OSC_VERSION "${GIT_BRANCH}-g${GIT_HASH}")
+	set(GIT_VERSION "-DGIT_VERSION=\"${GIT_VERSION}\"")
+	set(GIT_COMMIT_TIMESTAMP "-DGIT_COMMIT_TIMESTAMP=${GIT_COMMIT_TIMESTAMP}")
 else()
-	set(GIT_VERSION "")
-	set(GIT_COMMIT_TIMESTAMP 0)
 	set(OSC_VERSION v${PROJECT_VERSION})
 endif()
 
@@ -70,8 +70,8 @@ link_directories(${CMAKE_PREFIX_PATH}/lib)
 
 add_definitions(-Dlinux -D_GNU_SOURCE 
 	-DOSC_VERSION="${OSC_VERSION}"
-	-DGIT_VERSION="${GIT_VERSION}"
-	-DGIT_COMMIT_TIMESTAMP=${GIT_COMMIT_TIMESTAMP}
+	${GIT_VERSION}
+	${GIT_COMMIT_TIMESTAMP}
 	-Wall -Werror
 )
 

--- a/Makefile
+++ b/Makefile
@@ -12,10 +12,15 @@ CC := $(CROSS_COMPILE)gcc
 SYSROOT := $(shell $(CC) -print-sysroot)
 MULTIARCH := $(shell $(CC) -print-multiarch)
 
+ifneq ($(wildcard .git/*),)
 GIT_BRANCH := $(shell git name-rev --name-only HEAD | sed 's:.*/::')
 GIT_HASH := $(shell git describe --abbrev=7 --dirty --always)
-GIT_VERSION := $(shell git rev-parse --short HEAD)
-GIT_COMMIT_TIMESTAMP := $(shell git show -s --pretty=format:"%ct" HEAD)
+GIT_VERSION := -DGIT_VERSION=\"$(shell git rev-parse --short HEAD)\"
+GIT_COMMIT_TIMESTAMP := -DGIT_COMMIT_TIMESTAMP='"$(shell git show -s --pretty=format:"%ct" HEAD)"'
+OSC_VERSION=$(GIT_BRANCH)-g$(GIT_HASH)
+else
+OSC_VERSION=0.99
+endif
 
 WITH_MINGW := $(if $(shell echo | $(CC) -dM -E - |grep __MINGW32__),y)
 EXPORT_SYMBOLS := -Wl,--export-all-symbols
@@ -57,9 +62,9 @@ CFLAGS := $(DEP_CFLAGS) \
 	-Wextra -Wno-unused-parameter \
 	-Werror -g -std=gnu90 -D_GNU_SOURCE -O2 -funwind-tables \
 	-DPREFIX='"$(PREFIX)"' \
-	-DFRU_FILES=\"$(FRU_FILES)\" -DGIT_VERSION=\"$(GIT_VERSION)\" \
-	-DGIT_COMMIT_TIMESTAMP='"$(GIT_COMMIT_TIMESTAMP)"' \
-	-DOSC_VERSION=\"$(GIT_BRANCH)-g$(GIT_HASH)\" \
+	-DFRU_FILES=\"$(FRU_FILES)\" $(GIT_VERSION) \
+	$(GIT_COMMIT_TIMESTAMP) \
+	-DOSC_VERSION=\"$(OSC_VERSION)\" \
 	-D_POSIX_C_SOURCE=200809L \
 	-DSERIAL_BACKEND
 

--- a/dialogs.c
+++ b/dialogs.c
@@ -24,6 +24,10 @@
 #include "config.h"
 #include "phone_home.h"
 
+#ifndef GIT_VERSION
+#define GIT_VERSION	""
+#endif
+
 #ifdef SERIAL_BACKEND
 #include <libserialport.h> /* cross platform serial port lib */
 #endif
@@ -902,6 +906,7 @@ gint create_blocking_popup(GtkMessageType type, GtkButtonsType button,
 
 static glong date_compare_against_build_date(const char *iso8601_date)
 {
+#ifdef GIT_COMMIT_TIMESTAMP
 	GTimeVal time;
 	glong build_time = atol(GIT_COMMIT_TIMESTAMP);
 	gboolean parsed;
@@ -915,6 +920,9 @@ static glong date_compare_against_build_date(const char *iso8601_date)
 	}
 
 	return ret;
+#else
+	return 0;
+#endif
 }
 
 /*

